iothub/device/src/Twin/TwinProperties.cs

```diff
@@ -23,11 +25,23 @@ protected internal TwinProperties(DesiredProperties requestsFromService, Reporte
         /// <summary>
         /// The collection of desired property update requests received from service.
         /// </summary>
+        [JsonProperty("desiredProperties")]
```

Comment:
The json property name here should be the same as defined in the twin document, i.e. "desired", "reported".

For reference, that is what we define on the service client as well: https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/service/src/Twin/Models/ClientTwinDocument.cs